### ISSUE TITLE
Site Editor: drill-down menu 'Back' button focus color and missing tooltips

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#48211]
### Description

In the Site Editor, the'drill-down' on the left has now a 'Back' button that shows on all the levels.

- Visually, the button contains only an icon. As such, at the very least it should expose its accessible name in a tooltip. This is the standard for all the icon-only buttons.
- Ideally, the button accessible name should not change depending on the level, as the button visuals doesn't change.
- When the button is focused, the arrow color becomes dark blue and it's barely visible. This appears to happen only on the first level.

First level:
- The button icon becomes dark blue 😞 
- The button does show a tooltip with text 'Dashboard' 👍 
- Under the hood, the button accessible name doesn't match the visible tooltip text and it's `Navigate to the Dashboard` 😞  This is a barrier for speech recognition software users.

<img width="395" alt="Screenshot 2023-02-17 at 09 47 20" src="https://user-images.githubusercontent.com/1682452/219696599-792ffb84-d439-4d3a-a37e-08fb141119ef.png">

Second level:
- The button icon stays white 👍 
- The button does not show a tooltip 😞 
- Under the hood, the button accessible name changed to 'Back' 😞 (note: 'Back' is a good, short name. The point here is that it dynamically changed from the previous 'Navigate to the Dashboard' to 'Back'). This isn't great for speech recognition software users.

<img width="396" alt="Screenshot 2023-02-17 at 09 48 23" src="https://user-images.githubusercontent.com/1682452/219698708-43fd1068-928a-4ff6-9c70-a6b868007655.png">

Third level:
Same as the second level.

<img width="396" alt="Screenshot 2023-02-17 at 09 49 34" src="https://user-images.githubusercontent.com/1682452/219698966-8b6c608d-74d7-4dbb-ba11-f3d3dcb48010.png">

### Step-by-step reproduction instructions

- Go to the Site Editor.
- Navigate through the levels in the 'drill-down' menu.
- Check the icon color and inspect the source to confirm the issues described above.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
> First level:
> The button icon becomes dark blue 😞

This point will be resolved by #48134. However, the tests fail, probably due to problems with [the performance tests](https://github.com/WordPress/gutenberg/actions/workflows/performance.yml) themselves, so they have not yet been merged.
